nulib/meadow
# Add GraphQL query for DC API superuser token

# Summary 
Add GraphQL query for DC API superuser token

# Specific Changes in this PR
- Add `api_token` schema type
- Add `ApiToken` query type
- Add Terraform code to add API secret to Meadow config (reading directly from the deployed API CloudFormation stack)

# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.


# Steps to Test

1. Open to the GraphQL Playground (`/api/graphiql`)
2. Submit the following query:
    ```
    query{
      dcApiToken{
        expires
        token
      }
    }
    ```
3. `expires` should be about 5 minutes from now

Bonus token verification:

Using `curl` or Postman, send a request to the API's `/auth/whoami` route with the header `Authorization: Bearer [TOKEN_FROM_GRAPHQL_RESPONSE]`

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
- Backwards-incompatible API changes
- [x] Terraform changes (already applied on staging)


# Tested/Verified
